cli/src/pcluster/config/cluster_config.py

```diff
@@ -1645,7 +1645,10 @@ def _register_validators(self, context: ValidatorContext = None):  # noqa: D102
             volume_iops=root_volume.iops,
         )
         self._register_validator(KeyPairValidator, key_name=self.head_node.ssh.key_name, os=self.image.os)
-        self._register_validator(SchedulerDisableSudoAccessForDefaultUserValidator, scheduler=self.scheduling.scheduler)
+        if self.disable_sudo_access_default_user:
```

Comment:
*Clean Coding: separation of responsibilities*
For a better division of responsibilities, we should always register here the validator without any condition and delegate to the validator the responsibility to fail if the disablement is true and the scheduler is unsupported.